docs/recipes/MigratingToI18Next.md

```diff
@@ -0,0 +1,164 @@
+---
+title: Migrating from i18n-js to react-i18next
+description: How to migrate from i18n-js to react-i18next
+tags:
+  - i18n
+last_update:
+  author: Felipe Peña
+publish_date: 2024-09-25
+---
+
+# Migrating from i18n-js to react-i18next within an Ignite project
+
+## Overview
+
+In this guide, we will be going through the steps required to migrate your Ignite generated project from using `i18n-js` to the `react-i18next` library. It is meant specifically for projects generated with version 9 and below.
+
+`react-i18next` will be included in Ignite's version 10. If you're using an earlier version, this guide provides the necessary steps to successfully complete the migration.
+
+Finally, the steps outlined in this guide, are based on the changes outlined in the following two PRs:
+* [Swap out i18n-js for react-18next](https://github.com/infinitered/ignite/pull/2770)
+* [Fix language switching and update date-fns to v4](https://github.com/infinitered/ignite/pull/2778)
+
+## Step 1: Manage dependencies
+
+Remove the `i18n-js` package and its types:
+
+```bash
+yarn remove i18n-js
+yarn remove @types/i18n-js
+```
+
+Add the two new dependencies:
+
+```bash
+yarn add react-i18nex i18next
+```
+
+## Step 2: Set up initialization logic in app.tsx
+
+To ensure that `react-i18next` finishes initializing before your app proceeds, we recommend adding state and logic to the `app.tsx` entry file:
+
+1. Create a state variable, `isI18nInitialized`, to track initialization status.
+2. Use the `useEffect` hook to set the state when initialization completes.
+
+```js
+const [isI18nInitialized, setIsI18nInitialized] = useState(false);
+
+useEffect(() => {
+  initI18n().then(() => setIsI18nInitialized(true));
+}, []);
+```
+
+Additionally, consider including the new state variable in the rendering condition for the app.
+
+```js
+// error-line-start
+if (!rehydrated || !isNavigationStateRestored || (!areFontsLoaded && !fontLoadError)) {
+// error-line-end
+// success-line-start
+if (!rehydrated || !isNavigationStateRestored || !isI18nInitialized || (!areFontsLoaded && !fontLoadError)) {
+// success-line-end
+  return null
+}
+```
+
+This ensures that your app will wait until `react-i18next` is fully initialized before continuing, preventing any issues with missing translations.
+
+## Step 3: Update the i18n initialization method
+
+Next, update your i18n initialization to use `react-i18next`, which also includes RTL (right-to-left) language support and handles locale selection. In a Ingnite generated project, this is located in `app/i18n/i18n.ts`.
```

Comment:
Removal of `import { I18n } from "i18n-js"` might be a helpfuls step to add prior to bringing in